MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PolygonSelectToggleActivity.kt

```diff
@@ -125,13 +125,15 @@ class PolygonSelectToggleActivity : AppCompatActivity(), MapboxMap.OnMapClickLis
      */
     private fun handleClickIcon(screenPoint: PointF): Boolean {
         val features = mapboxMap.queryRenderedFeatures(screenPoint, BASE_NEIGHBORHOOD_FILL_LAYER_ID)
-        val name = features[0].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY)
-        val featureList = featureCollection?.features()
-        (0 until featureList!!.size).iterator().forEach {
-            if (featureList[it].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY) == name) {
-                if (featureSelectStatus(it))
-                    setFeatureSelectState(featureList[it], false) else
-                    setSelected(it)
+        if (features.isNotEmpty()) {
+            val name = features[0].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY)
+            val featureList = featureCollection?.features()
+            (0 until featureList!!.size).iterator().forEach {
```

Comment:
nit `featureList!!` doesn't look safe, even though logically the featureCollection shouldn't be null after `queryingRenderedFeatures` returned non empty list.

Reply:
Fixed 👍